Modulo3/lista-typescript/src/exercicio11.ts

```diff
@@ -0,0 +1,173 @@
+
+const number: string = prompt("um numero de 1 a 1990")
+
+const NumerosConv:any|string[] = {
+    
+    0 : [
+         ''		
+        ,'I' 	
+        ,'II'	
+        ,'III'	
+        ,'IV'	
+        ,'V'	
+        ,'VI'	
+        ,'VII'	
+        ,'VIII'	
+        ,'IX'					
+    ]
+    
+    ,1 : [
+         ''		
+        ,'X'	
+        ,'XX'   
+        ,'XXX'  
+        ,'XL'   
+        ,'L'    
+        ,'LX'   
+        ,'LXX'  
+        ,'LXXX' 
+        ,'XC'   
+    ]
+    ,2 : [
+        ''
+        ,'C'	
+        ,'CC'   
+        ,'CCC'  				
+        ,'CD'   
+        ,'D'    
+        ,'DC'   
+        ,'DCC'  
+        ,'DCCC' 
+        ,'CM' 	
+    ]
+    ,3 : [
+         ''
+        ,'M' 
+    ]
+};
+
+
+function RetornaNumero(number:string){		
+	
+    number = number.replace(/[^0-9]/, "");
+    if(number == '') return'';
+    
+    const Numero = parseInt(number);
+    
+    if(Numero > 1999){
+        return "número maximo '1999'";
+    }
+    
+
+   
+    const ordNumero = Number(number).toString();
+    const OrLength = ordNumero.length;
+    
+    
+    const unidadeDezenaCentena = OrLength - 1;
+    
+    
+    let NovaOrd = '';
+    for(const i = unidadeDezenaCentena; i >= 0 ;i--){
+        NovaOrd = NovaOrd + ordNumero.charAt(i);
+    }
+    
+    let ReturnN = '';
+    for(const i = unidadeDezenaCentena; i >= 0 ;i--){
```

Comment:
const não muda, o i dentro do for, vai sofre alterações, por isso usamos o let para declarar o índice no for.